### PR TITLE
Bugfixing

### PR DESCRIPTION
**Pull request by pabloalba**
Fixes
https://tree.taiga.io/project/penpot/issue/6255
https://tree.taiga.io/project/penpot/issue/6019
https://tree.taiga.io/project/penpot/issue/6030
https://tree.taiga.io/project/penpot/issue/6013
https://tree.taiga.io/project/penpot/issue/6282
https://tree.taiga.io/project/penpot/issue/6288

### CODE REVIEW

File: frontend/src/app/main/data/workspace/libraries.cljs [hunk 1]
```diff
@@ -944,7 +944,20 @@
                                  (when sync-typographies?
                                    (dwlh/generate-sync-file it file-id :typographies asset-id library-id state))])
 
-               changes         (pcb/concat-changes library-changes file-changes)]
+               changes         (pcb/concat-changes library-changes file-changes)
+
+               find-shape      (fn [data]
```
Comment (hirunatan) [inline frontend/src/app/main/data/workspace/libraries.cljs]: We should add a rule that forbids naming a parameter `data` :grimacing:.

When I read this code, i have absolutely no clue of what kind of data comes here... If this is a change, the name should be `change`. When we have not data types, at least the name helps us to understand what data are we talking about.

Comment (hirunatan) [reply frontend/src/app/main/data/workspace/libraries.cljs]: Also the name `find-shape` suggests that it returns one shape, but it actually returns a list, don't it?

File: common/src/app/common/files/changes.cljc [hunk 2]
```diff
@@ -477,12 +477,16 @@
           ;; or inside its main component if it's in a copy,
           ;; or inside a copy
           (is-valid-move? [objects shape-id]
-            (let [invalid-targets (calculate-invalid-targets objects shape-id)]
-              (and (contains? objects shape-id)
+            (let [invalid-targets (calculate-invalid-targets objects shape-id)
+                  shape (get objects shape-id)
+                  _ (prn "is-valid-move?")]
```
Comment (hirunatan) [inline common/src/app/common/files/changes.cljc]: prn

File: frontend/src/app/main/data/workspace/libraries.cljs (hunk 1 above)
Comment (pabloalba) [reply frontend/src/app/main/data/workspace/libraries.cljs]: Fixed

**Comment by niwinz** [comment]
image
Commit message :-1:

**Pull request merged**